\documentclass[11pt]{article}
\usepackage{multirow}
\usepackage{hyperref}
\usepackage{xr}
\usepackage{listings}
\usepackage{algorithm}
\usepackage{algpseudocode}
\usepackage[export]{adjustbox}
\usepackage{float}
\usepackage[caption = false]{subfig}
\usepackage{graphicx}
%\usepackage{lineno}
\input{macros}

\externaldocument{supplement}

\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm


%The next command sets up an environment for the abstract to your paper.

\newenvironment{sciabstract}{%
\begin{quote} \bf}
{\end{quote}}

\def\VB#1{{\textcolor{blue}{VB note: #1}}}


% If your reference list includes text notes as well as references,
% include the following line; otherwise, comment it out.

\renewcommand\refname{References}

% The following lines set up an environment for the last note in the
% reference list, which commonly includes acknowledgments of funding,
% help, etc.  It's intended for users of BibTeX or the {thebibliography}
% environment.  Users who are hand-coding their references at the end
% using a list environment such as {enumerate} can simply add another
% item at the end, and it will be numbered automatically.

%\newcounter{lastnote}
%\newenvironment{scilastnote}{%
%\setcounter{lastnote}{\value{enumiv}}%
%\addtocounter{lastnote}{+1}%
%\begin{list}%
%{\arabic{lastnote}.}
%{\setlength{\leftmargin}{.22in}}
%{\setlength{\labelsep}{.5em}}}
%{\end{list}}


% Include your paper's title here

\title{ViFi: Accurate detection of viral integration and mRNA fusion reveals transcription of non-coding regions in cervical cancer} 

\author
{Nam-phuong D. Nguyen$^{1}$, Viraj Deshpande$^{1}$, 
Jens Luebeck$^{2}$, Paul S. Mischel$^{3,4,5}$, and Vineet Bafna$^{1}$\\
\normalsize{$^{1}$Computer Science and Engineering, University of California,}\\
\normalsize{San Diego, La Jolla, 92093, USA.,}\\
\normalsize{$^{2}$Bioinformatics and Systems Biology Program, University of California,}\\
\normalsize{San Diego, La Jolla, 92093, USA.,}\\
\normalsize{$^{3}$Ludwig Institute for Cancer Research, University of California,}\\
\normalsize{San Diego, La Jolla, 92093, USA.,}\\
\normalsize{$^{4}$Department of Pathology, University of California,}\\
\normalsize{San Diego, La Jolla, 92093, USA.,}\\
\normalsize{$^{5}$Moores Cancer Center, University of California,}\\
\normalsize{San Diego, La Jolla, 92093, USA.}\\
%\normalsize{$^\ast$To whom correspondence should be addressed; E-mail:  vbafna@eng.ucsd.edu.}
}

% Include the date command, but leave its argument blank.

\date{}

%%%%%%%%%%%%%%%%% END OF PREAMBLE %%%%%%%%%%%%%%%%

\externaldocument{supplement}

\begin{document} 

% Double-space the manuscript.

\baselineskip24pt

% Make the title.

\maketitle 

% Place your abstract within the special {sciabstract} environment.
%\linenumbers
\input{abstract.txt}
\input{introduction.txt}
\section*{Results}
\input{results_simulation.txt}
\input{results_bio.txt}
\input{discussion.txt}
\input{methods.txt}

%One possible direction to better understand the role of LINE and LTR expression is combining proteomics and genomics.  In particular, focusing on whether HPV positive samples with integration show higher expression of L1 encoded proteins, ORF1p and ORF2p, compared to HPV positive samples without integration.  

\section*{Acknowledgements}
This research was supported in part by grants from the NSF (DBI-1458557), and NIH (R01GM114362).  This work used the Extreme Science and Engineering Discovery Environment (XSEDE), which is supported by National Science Foundation grant number ACI-1053575.  XSEDE resources were provided by TG-ASC160042.  

The results published here are in whole or part based upon data generated by the TCGA Research Network: \href{http://cancergenome.nih.gov/}{http://cancergenome.nih.gov/}.

%\section*{Author Information}
%\subsection*{Affiliations}
%\textbf{Computer Science and Engineering, University of California, San Diego, La Jolla, 92093, USA}\\
%Nam-phuong Nguyen, Viraj Deshpande, and Vineet Bafna\\
%\textbf{Bioinformatics and Systems Biology Program, University of California, San Diego, La Jolla, 92093, USA}\\
%Jens Luebeck\\
%\textbf{Ludwig Institute for Cancer Research, San Diego, La Jolla, 92093, USA}\\
%Paul S. Mischel\\
%\textbf{Department of Pathology, University of California, San Diego, La Jolla, 92093, USA}\\
%Paul S. Mischel\\
%\textbf{Moores Cancer Center, University of California, San Diego, La Jolla, 92093, USA}\\
%Paul S. Mischel\\

\subsection*{Author contributions}

N.N. and V.B. conceived the study and designed the experiment.  N.N. conducted the experiments.  J.L. prepared the simulated data.  N.N. and V.D. designed and coded ViFi.  N.N., V.D., J.L., P.M and V.B. analyzed the data.  N.N., V.B., and P.M. drafted the manuscript with contributions from V.D. and J. L.  


\section*{Disclosure Declaration}
V.B. is a co-founder, has an equity interest, and receives income from Digital Proteomics, LLC. The terms of this arrangement have been reviewed and approved by the University of California, San Diego in accordance with its conflict of interest policies. DP was not involved in the research presented here.

%\section*{Corresponding author}
%Correspondence to Vineet Bafna and Paul S. Mischel.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%
%\bibliographystyle{naturemag}
\bibliographystyle{genres}
%
%\bibliography{Document}

%\bibliographystyle{natbib}
\def\bibfont{\normalsize}
\bibliography{main}
%\input{supplement.tex}
\end{document}

